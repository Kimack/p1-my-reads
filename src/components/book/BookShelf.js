import React from 'react'
import BookList from './BookList.js'
import PropTypes from 'prop-types';

const BookShelf = ({ shelf, books, shelves, onBookShelfChange }) => (
    <div className="bookshelf">
        <h2 className="bookshelf-title">{shelf.title}({books.length})</h2>
        <div className="bookshelf-books">
            <BookList
                books={books}
                shelves={shelves}
                onBookShelfChange={onBookShelfChange} />
        </div>
    </div>
)

BookShelf.propTypes = {
    /**
     * Current shelf
     */
    shelf: PropTypes.object.isRequired,
	/**
	 * Array of current books on any shelf
	 */
	books: PropTypes.array.isRequired,
	/**
	 * Available shelves array
	 */
    shelves: PropTypes.array.isRequired,
	/**
	 * Function that will be called when the shelf is changed
	 */
	onBookShelfChange: PropTypes.func.isRequired
}

export default BookShelf;