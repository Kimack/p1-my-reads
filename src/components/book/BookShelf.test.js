import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import BookShelf from './BookShelf';

configure({ adapter: new Adapter() });

describe('BookList instance is rendered corretly', () => {
    it('renders without crashing', () => {
        const component = <BookShelf shelf={{ title: "Shelft Title" }} books={[{ id: 0 }, { id: 1 }]} shelves={[]} onBookShelfChange={() => { }} />;
        expect(shallow(component)).toMatchSnapshot();
    });
});