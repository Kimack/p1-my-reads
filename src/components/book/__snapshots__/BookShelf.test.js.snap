// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList instance is rendered corretly renders without crashing 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Shelft Title
    (
    2
    )
  </h2>
  <div
    className="bookshelf-books"
  >
    <BookList
      books={
        Array [
          Object {
            "id": 0,
          },
          Object {
            "id": 1,
          },
        ]
      }
      onBookShelfChange={[Function]}
      shelves={Array []}
    />
  </div>
</div>
`;
